src/pyhf/cli/patchset.py

```diff
@@ -0,0 +1,161 @@
+"""The pyhf spec CLI subcommand."""
+import logging
+
+import click
+import json
+import sys
+import jsonpatch
+
+from ..workspace import Workspace
+from .. import utils
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+
+@click.group(name='patchset')
+def cli():
+    """Operations involving patchsets."""
+
+
+@cli.command()
+@click.argument('patchset', default='-')
+@click.option(
+    '--name', help='The name of the patch to extract.', default=None,
+)
+@click.option(
+    '--output-file',
+    help='The location of the output json file. If not specified, prints to screen.',
+    default=None,
+)
+@click.option(
+    '--with-metadata/--without-metadata',
+    default=False,
+    help="Include patchset metadata in output.",
+)
+def extract(patchset, name, output_file, with_metadata):
+    """
+    Extract a patch from a patchset.
+
+    If the patchset does not contain the name, will exit with code 1.
+
+    Returns:
+        jsonpatch (:obj:`list`): A list of jsonpatch operations to apply to a workspace.
+    """
+    with click.open_file(patchset, 'r') as fstream:
+        patchset_obj = json.load(fstream)
+
+    utils.validate(patchset_obj, 'patchset.json')
+
+    extracted_patch_obj = None
+    for patch in patchset_obj['patches']:
+        if patch['name'] == name:
```

Comment:
how do we ensure there is max 1 point with that name in the patchset.

Reply:
I'm thinking of adding a `Patchset()` class that will do some of these checks for us -- because it's the same problem we had with the `Workspace` / `Model` stuff. This should also simplify / DRY a lot of the code here.